pkgs/development/interpreters/guile/1.8.nix

```
@@ -1,32 +1,51 @@
-{ lib, stdenv, pkgsBuildBuild, buildPackages
-, fetchurl, makeWrapper, gawk, pkg-config
-, libtool, readline, gmp
+{ lib
+, stdenv
+, fetchurl
+, buildPackages
+, gawk
+, gmp
+, libtool
+, makeWrapper
+, pkg-config
+, pkgsBuildBuild
+, readline
 }:
 
 stdenv.mkDerivation rec {
-  name = "guile-1.8.8";
+  pname = "guile";
+  version = "1.8.8";
 
   src = fetchurl {
-    url = "mirror://gnu/guile/${name}.tar.gz";
+    url = "mirror://gnu/${pname}/${pname}-${version}.tar.gz";
```

```suggestion
    url = "mirror://gnu/guile/guile-${version}.tar.gz";
```
Not replacing every possible occurrence of pname is preferred. Otherwise it is harder than it needs to be to get the original URL.

---

Not even `${pname}-${version}.tgz`?